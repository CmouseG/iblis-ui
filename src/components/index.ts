export { //
    ChangePasswordForm, ChangePasswordFormProps, ChangePasswordFormValues, //
    DeleteAccountForm, DeleteAccountFormProps, DeleteAccountFormValues, //
    ManagedAddSimpleForm, ManagedAddSimpleFormProps, ManagedAddSimpleFormValues, //
    ManagedSimpleForm, ManagedSimpleFormData, ManagedSimpleFormProps, //
    ManagedSimpleListForm, ManagedSimpleListFormProps, ManagedSimpleListFormValues, //
    UserProfileForm, UserProfileFormProps, UserProfileFormValues, //
    WizardForm //
} from './form_components';
export {
    AccountCard, AccountCardProps, //
    AccountMenu, AccountMenuLevelProps, AccountMenuProps, //
    EmptyPage, //
    FilterButton, FilterButtonMenuProps, FilterButtonProps, //
    FooterMenu, FooterMenuLevelProps, FooterMenuProps, //
    HelperButton, HelperButtonProps, //
    IblisButton, IblisButtonProps, //
    IblisSelectField, IblisSelectFieldProps, //
    IblisSnackbar, IblisSnackbarProps, //
    IblisTextField, IblisTextFieldProps, //
    MainMenu, MainMenuLevelProps, MainMenuProps, //
    ProgressBar, ProgressBarProps, //
    UploadAvatar, UploadAvatarProps
} from './ui_components';
