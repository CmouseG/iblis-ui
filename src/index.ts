export {
    AccountCard, AccountCardProps, //
    AccountMenu, AccountMenuLevelProps, AccountMenuProps, //
    ChangePasswordForm, ChangePasswordFormProps, ChangePasswordFormValues, //
    ConfirmSignupForm, ConfirmSignupFormProps, ConfirmSignupFormValues, //
    DeleteAccountForm, DeleteAccountFormProps, DeleteAccountFormValues, //
    EmptyPage, //
    FilterButton, FilterButtonMenuProps, FilterButtonProps, //
    FooterMenu, FooterMenuLevelProps, FooterMenuProps, //
    ForceChangePasswordForm, ForceChangePasswordFormProps, ForceChangePasswordFormValues, //
    HelperButton, HelperButtonProps, //
    IblisButton, IblisButtonProps, //
    IblisSelectField, IblisSelectFieldProps, //
    IblisSnackbar, IblisSnackbarProps, IblisTextField, IblisTextFieldProps, //
    LoginForm, LoginFormProps, LoginFormValues, //
    MainMenu, MainMenuLevelProps, MainMenuProps, //
    ManagedAddSimpleForm, ManagedAddSimpleFormProps, ManagedAddSimpleFormValues, //
    ManagedSimpleForm, ManagedSimpleFormData, ManagedSimpleFormProps, //
    ManagedSimpleListForm, ManagedSimpleListFormProps, ManagedSimpleListFormValues, //
    PasswordConfirmForm, PasswordConfirmFormProps, PasswordConfirmFormValues, //
    ProgressBar, ProgressBarProps, //
    ResetPasswordForm, ResetPasswordFormProps, ResetPasswordFormValues, //
    SignupForm, SignupFormProps, SignupFormValues, //
    StripeForm, StripeFormProps, //
    UploadAvatar, UploadAvatarProps, //
    UserProfileForm, UserProfileFormProps, UserProfileFormValues //
} from './components';
export { //
    CardMenu, CardMenuLevelProps, CardMenuProps, //
    CardTitleLayout, CardTitleLayoutProps, //
    ContentWithTitle, ContentWithTitleProps, //
    ExternalFormLayout, ExternalFormLayoutProps, //
    MainLayout, MainLayoutProps
} from './layouts';
export { theme } from './theme';
// tslint:disable-next-line:no-import-side-effect
import 'iblis-font/index.css';
// tslint:disable-next-line:no-import-side-effect
import 'typeface-roboto/index.css';
