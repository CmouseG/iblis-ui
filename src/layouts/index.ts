export { CardMenu, CardMenuLevelProps, CardMenuProps } from './card_menu';
export { CardTitleLayout, CardTitleLayoutProps } from './card_title_layout';
export { ContentWithTitle, ContentWithTitleProps } from './content_with_title';
export { ExternalFormLayout, ExternalFormLayoutProps } from './external_form_layout';
export { MainLayout, MainLayoutProps } from './main_layout';
