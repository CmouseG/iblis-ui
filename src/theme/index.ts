export { theme } from './iblis_theme';
