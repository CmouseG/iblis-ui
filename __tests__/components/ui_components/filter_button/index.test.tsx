import * as React from 'react';
import renderer from 'react-test-renderer';
import { FilterButton, FilterButtonMenuProps } from '../../../../src';

it('renders correctly', () => {
    const filterMenu: FilterButtonMenuProps[] = [
        { name: 'All', action: () => undefined, count: '88' },
        { name: 'Published', action: () => undefined, count: '34' },
        { name: 'Draft', action: () => undefined, count: '12' },
    ];
    const tree = renderer
        .create(
            <FilterButton
                listFilter={filterMenu}
                activeIndex={0}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
