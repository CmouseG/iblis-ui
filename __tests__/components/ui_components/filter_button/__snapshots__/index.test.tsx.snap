// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiGrid-item-2"
>
  <button
    className="MuiButtonBase-root-98"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "background": "#424242",
        "border": "1px solid#757575",
        "borderBottomLeftRadius": "6px",
        "borderBottomRightRadius": 0,
        "borderTopLeftRadius": "6px",
        "borderTopRightRadius": 0,
        "lineHeight": "0",
        "padding": 8,
        "paddingBottom": 6,
      }
    }
    tabIndex="0"
    type="button"
  >
    <p
      className="MuiTypography-root-101 MuiTypography-body1-110 MuiTypography-alignCenter-114"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      All
    </p>
    <span
      className="MuiTypography-root-101 MuiTypography-caption-111 MuiTypography-alignCenter-114"
      style={
        Object {
          "paddingLeft": 8,
        }
      }
    >
      88
    </span>
    <span
      className="MuiTouchRipple-root-126"
    />
  </button>
  <button
    className="MuiButtonBase-root-98"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "borderBottom": "1px solid#757575",
        "borderRadius": 0,
        "borderRight": "1px solid#757575",
        "borderTop": "1px solid#757575",
        "padding": 8,
        "paddingBottom": 6,
      }
    }
    tabIndex="0"
    type="button"
  >
    <p
      className="MuiTypography-root-101 MuiTypography-body1-110 MuiTypography-alignCenter-114"
    >
      Published
    </p>
    <span
      className="MuiTypography-root-101 MuiTypography-caption-111 MuiTypography-alignCenter-114"
      style={
        Object {
          "paddingLeft": 8,
        }
      }
    >
      34
    </span>
    <span
      className="MuiTouchRipple-root-126"
    />
  </button>
  <button
    className="MuiButtonBase-root-98"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "borderBottom": "1px solid#757575",
        "borderBottomLeftRadius": 0,
        "borderBottomRightRadius": "6px",
        "borderRight": "1px solid#757575",
        "borderTop": "1px solid#757575",
        "borderTopLeftRadius": 0,
        "borderTopRightRadius": "6px",
        "padding": 8,
        "paddingBottom": 6,
      }
    }
    tabIndex="0"
    type="button"
  >
    <p
      className="MuiTypography-root-101 MuiTypography-body1-110 MuiTypography-alignCenter-114"
    >
      Draft
    </p>
    <span
      className="MuiTypography-root-101 MuiTypography-caption-111 MuiTypography-alignCenter-114"
      style={
        Object {
          "paddingLeft": 8,
        }
      }
    >
      12
    </span>
    <span
      className="MuiTouchRipple-root-126"
    />
  </button>
</div>
`;
